staging/src/k8s.io/apiserver/pkg/admission/plugin/cel/filter.go

```diff
@@ -0,0 +1,244 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cel
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/google/cel-go/interpreter"
+
+	admissionv1 "k8s.io/api/admission/v1"
+	authenticationv1 "k8s.io/api/authentication/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apiserver/pkg/admission"
+	"k8s.io/apiserver/pkg/admission/plugin/webhook/generic"
+)
+
+// filterCompiler implement the interface FilterCompiler.
+type filterCompiler struct {
+}
+
+func NewFilterCompiler() FilterCompiler {
+	return &filterCompiler{}
+}
+
+type evaluationActivation struct {
+	object, oldObject, params, request interface{}
+}
+
+// ResolveName returns a value from the activation by qualified name, or false if the name
+// could not be found.
+func (a *evaluationActivation) ResolveName(name string) (interface{}, bool) {
```

Comment:
activations and, to some degree, envs and compilers might not worth reusing.

Reply:
These are required for the cel evaluation to work. cel program.Eval casts to 

```
package interpreter

// Activation used to resolve identifiers by name and references by id.
//
// An Activation is the primary mechanism by which a caller supplies input into a CEL program.
type Activation interface {
	// ResolveName returns a value from the activation by qualified name, or false if the name
	// could not be found.
	ResolveName(name string) (interface{}, bool)

	// Parent returns the parent of the current activation, may be nil.
	// If non-nil, the parent will be searched during resolve calls.
	Parent() Activation
}
```

Reply:
I think you will end up with creating your own Activation and env. There are more variables coming to admission controller like `audit`. 

It is likely that you cannot reuse them. I recommend that these are kept where they are now.

Reply:
Oh you were asking about moving them my bad. So I moved all of the cel logic to this cel package, this is part of the cel evaluation logic. Its not used elsewhere. The goal is for the cel package to encapsulate the cel logic and this is required for that.

Reply:
This is a good point! In the current way of refactor, those will be shared within both path. It is not a issue for ValidatingAdmissionPolicy but might be something to consider in webhook match later since webhook does not need `params` as of now per my understanding? Also, `audit` will be added soon in ValidatingAdmissionPolicy which will add another variable not needed by Webhook. cc @jpbetz

Reply:
The plan for params is we can setup the compiler with params false and pass null params for webhooks

Reply:
But I see your point about the value not being required for them and being available